cmd/thv-registry-api/app/serve.go

```diff
@@ -0,0 +1,164 @@
+}
+
+const (
+	defaultGracefulTimeout = 30 * time.Second // Kubernetes-friendly shutdown time
+	serverRequestTimeout   = 10 * time.Second // Registry API should respond quickly
+	serverReadTimeout      = 10 * time.Second // Enough for headers and small requests
+	serverWriteTimeout     = 15 * time.Second // Must be > serverRequestTimeout to let middleware handle timeout
+	serverIdleTimeout      = 60 * time.Second // Keep connections alive for reuse
+)
+
+func init() {
+	serveCmd.Flags().String("address", ":8080", "Address to listen on")
+	serveCmd.Flags().String("configmap", "", "ConfigMap name containing registry data")
+
+	// Bind flags to viper
+	_ = viper.BindPFlag("address", serveCmd.Flags().Lookup("address"))
+	_ = viper.BindPFlag("configmap", serveCmd.Flags().Lookup("configmap"))
+	_ = viper.BindPFlag("namespace", serveCmd.Flags().Lookup("namespace"))
+}
+
+// getKubernetesConfig returns a Kubernetes REST config
+func getKubernetesConfig() (*rest.Config, error) {
+	// Try in-cluster config first
+	config, err := rest.InClusterConfig()
+	if err == nil {
+		return config, nil
+	}
+
+	// Fall back to kubeconfig
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+	return kubeConfig.ClientConfig()
+}
+
+func runServe(_ *cobra.Command, _ []string) error {
+	ctx := context.Background()
+
+	// Get configuration
+	address := viper.GetString("address")
+	configMapName := viper.GetString("configmap")
+
+	namespace := thvk8scli.GetCurrentNamespace()
+
+	logger.Infof("Starting registry API server on %s", address)
+	logger.Infof("ConfigMap: %s, Namespace: %s", configMapName, namespace)
+
+	// Create Kubernetes client and providers
+	var registryProvider service.RegistryDataProvider
+	var deploymentProvider service.DeploymentProvider
+
+	if configMapName != "" {
+		// Get Kubernetes config
+		config, err := getKubernetesConfig()
+		if err != nil {
+			return fmt.Errorf("failed to create kubernetes config: %w", err)
+		}
+
+		// Create Kubernetes client
+		clientset, err := kubernetes.NewForConfig(config)
+		if err != nil {
+			return fmt.Errorf("failed to create kubernetes client: %w", err)
+		}
+
+		// Create the Kubernetes-based registry data provider
+		registryProvider = service.NewK8sRegistryDataProvider(clientset, configMapName, namespace)
+		logger.Infof("Created Kubernetes registry data provider for ConfigMap %s/%s", namespace, configMapName)
+
+		// Create the Kubernetes-based deployment provider
+		deploymentProvider, err = service.NewK8sDeploymentProvider(config, configMapName)
+		if err != nil {
+			return fmt.Errorf("failed to create kubernetes deployment provider: %w", err)
+		}
+		logger.Infof("Created Kubernetes deployment provider for registry: %s", configMapName)
+	} else {
+		logger.Info("No ConfigMap specified, registry API will serve without registry data")
```

Comment:
logs here are confusing, it tells to serve without configmap, but we are erroring after it

Reply:
oof good catch thank you